Engine/Shared/EventListener/EventListener.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type
+** File description:
+** EventListener.cpp
+*/
+
+#include <iostream>
+
+#include "EventListener.hpp"
+
+EventListener::EventListener(std::shared_ptr<ECS::Registry> registry,
+                             std::shared_ptr <SceneManager::ISceneManager> sceneManager,
+                             std::shared_ptr <ABINetwork::INetworkUnit> networkUnit,
+                             std::shared_ptr <IGraphic> graphicLib)
+{
+    _registry = registry;
+    _sceneManager = sceneManager;
+    _networkUnit = networkUnit;
+    _graphicLib = graphicLib;
```

Comment:
Check if not null.

Reply:
Since the scene manager isn't sent at creation but just after (caused by a include loop) it can be nullptr and the graphic library can be nullptr too in the server side, i'll only check the registry and the networkUnit